vividus-plugin-applitools/src/main/java/org/vividus/visual/eyes/service/ImageVisualTestingService.java

```diff
@@ -93,14 +96,22 @@ private void setImages(String readKey, ApplitoolsVisualCheckResult visualCheckRe
 
     private String getImageAsBase64(String url, String readKey)
     {
-        if (url != null)
+        if (url != null && !apiUnauthorized)
         {
             URI imageUrl = URI.create(addKeyToUrl(url, readKey));
             try
             {
                 HttpResponse response = httpClient.doHttpGet(imageUrl);
                 byte[] body = response.getResponseBody();
-                if (response.getStatusCode() == HttpStatus.SC_OK && body != null)
+                int statusCode = response.getStatusCode();
+                if (statusCode == HttpStatus.SC_UNAUTHORIZED)
+                {
+                    apiUnauthorized = true;
+                    LOGGER.warn("The \"readApiKey\" property is not set or incorrect. "
```

Comment:
Does the body contain any meaningful information?

Reply:
No, in cases when API returns 401 SC body is empty